stl/inc/xutility

```diff
@@ -1680,6 +1680,329 @@ _NODISCARD constexpr const _Elem* data(initializer_list<_Elem> _Ilist) noexcept
+        noexcept(_Fake_copy_init<bool>(_Left < _Right._Current))) /* strengthened */ {
+        return _Left < _Right._Current;
+    }
+
+    template <_Not_bci_order<_Iter> _Other>
+    _NODISCARD_FRIEND constexpr bool operator>(const _Other& _Left, const basic_const_iterator& _Right) noexcept(
+        noexcept(_Fake_copy_init<bool>(_Left > _Right._Current))) /* strengthened */ {
+        return _Left > _Right._Current;
+    }
+
+    template <_Not_bci_order<_Iter> _Other>
+    _NODISCARD_FRIEND constexpr bool operator<=(const _Other& _Left, const basic_const_iterator& _Right) noexcept(
+        noexcept(_Fake_copy_init<bool>(_Left <= _Right._Current))) /* strengthened */ {
+        return _Left <= _Right._Current;
+    }
+
+    template <_Not_bci_order<_Iter> _Other>
+    _NODISCARD_FRIEND constexpr bool operator>=(const _Other& _Left, const basic_const_iterator& _Right) noexcept(
+        noexcept(_Fake_copy_init<bool>(_Left >= _Right._Current))) /* strengthened */ {
+        return _Left >= _Right._Current;
+    }
+
+    template <_Bci_order_3way<_Iter> _Other>
+    _NODISCARD_FRIEND constexpr auto operator<=>(const basic_const_iterator& _Left, const _Other& _Right) noexcept(
+        noexcept(_Left._Current <=> _Right)) /* strengthened */ {
+        return _Left._Current <=> _Right;
+    }
+
+    _NODISCARD_FRIEND constexpr basic_const_iterator operator+(const basic_const_iterator& _It,
+        const difference_type _Off) noexcept(noexcept(basic_const_iterator{_It._Current + _Off})) // strengthened
+        requires random_access_iterator<_Iter> {
+        return basic_const_iterator{_It._Current + _Off};
+    }
+
+    _NODISCARD_FRIEND constexpr basic_const_iterator operator+(const difference_type _Off,
+        const basic_const_iterator& _It) noexcept(noexcept(basic_const_iterator{_It._Current + _Off})) // strengthened
+        requires random_access_iterator<_Iter> {
+        return basic_const_iterator{_It._Current + _Off};
+    }
+
+    _NODISCARD_FRIEND constexpr basic_const_iterator operator-(const basic_const_iterator& _It,
+        const difference_type _Off) noexcept(noexcept(basic_const_iterator{_It._Current - _Off})) // strengthened
+        requires random_access_iterator<_Iter> {
+        return basic_const_iterator{_It._Current - _Off};
+    }
+
+    template <sized_sentinel_for<_Iter> _Sent>
+    _NODISCARD constexpr difference_type operator-(const _Sent& _Se) const // per LWG-3769
+        noexcept(noexcept(_Current - _Se)) /* strengthened */ {
+        return _Current - _Se;
+    }
+
+    // clang-format off
+    template <_Not_a_const_iterator _Sent>
+        requires sized_sentinel_for<_Sent, _Iter>
+    _NODISCARD_FRIEND constexpr difference_type operator-(const _Sent& _Se, const basic_const_iterator& _It) noexcept(
+        noexcept(_Se - _It.base())) /* strengthened */ { // per LWG-3769
+        // clang-format on
+        return _Se - _It.base();

```

Comment:
Is there a reason that this uses `_Se - _It.base()` instead of `_Se - _It._Current`?

Reply:
No -> fixed.

There used to be reason - I've made this function non-member in [`57c17`](https://github.com/microsoft/STL/pull/3043/commits/57c17d765733ae1da85ec1372af9876a600a2da7), so I had to use `base()` instead of `_Current` (and later @CaseyCarter implemented [LWG-3769](https://cplusplus.github.io/LWG/issue3769) without changing it back).